app/controllers/locations/map_controller.rb

```diff
@@ -0,0 +1,23 @@
+require_dependency "locations/application_controller"
+
+module Locations
+  class MapController < ApplicationController
+
+    def index
+        @map = Map.new(map_params)
+
+        if !@map.url.nil?
+          redirect_to @map.url
+        else
+          render plain: "Invalid parameters.", status: 404
+        end
+
+    end
+
+    # Only allow a trusted parameter "white list" through.
+    def map_params
+      params.permit(:id, :loc)
```

Comment:
If you want to use validates in the model you need to include ```include ActiveModel::Validations``` since this model isn't descended from active record.